lsp/source/served/lsp/jsonrpc.d

```diff
@@ -225,13 +225,13 @@ class RPCProcessor : Fiber
 	void registerCapability(T)(scope const(char)[] id, scope const(char)[] method, T options)
 	{
 		const(char)[][7] parts = [
-			`{"jsonrpc":"2.0","method":"client/registerCapability","registrations":[{"id":"`,
+			`{"jsonrpc":"2.0","method":"client/registerCapability","params":{"registrations":[{"id":"`,
 			id.escapeJsonStringContent,
 			`","method":"`,
 			method.escapeJsonStringContent,
 			`","registerOptions":`,
 			options.serializeJson,
-			`]}`
```

Comment:
For me just closing curly bracket is enough to make coc.nvim work.

Reply:
I have just compared the result of registerCapability on:

https://github.com/Pure-D/serve-d/commit/4fb7a5546acd6e7b99eb85a1ae6354ce72851409 (last working version with coc)
`
{"id":"ed99ff5c-eab0-405d-835f-5452bb7394be","jsonrpc":"2.0","method":"client/registerCapability","params":{"registrations":[{"id":"profilegc.watchfiles","method":"workspace/didChangeWatchedFiles","registerOptions":{"watchers":[{"globPattern":"**/profilegc.log"}]}}]}}`

and 

current master:
` {"jsonrpc":"2.0","method":"client/registerCapability","registrations":[{"id":"profilegc.watchfiles","method":"workspace/didChangeWatchedFiles","registerOptions":{"watchers":[{"globPattern":"**/profilegc.log"}]}]}`


I will say that it's more correct having params, but not required as you have detected.

Reply:
the last closing bracket is definitely required here, as done in the PR right now. If it works without it, there is probably a bug in the parsing code on the client, because that would still be invalid JSON.